processing/src/utils/minio_utils.py

```diff
@@ -1,9 +1,62 @@
 from minio import Minio
+from minio.credentials import AWSConfigProvider, EnvAWSProvider, IamAwsProvider
 
 from src.config import settings
 from src.utils.logger import get_logger
 
-logger = get_logger(__file__)
+logger = get_logger(__name__)
+
+
+class NotConfiguredException(Exception):
+    pass
+
+
+def create_minio_config():
+    minio_config = {}
+
+    minio_config.update({"secure": settings.minio_secure_connection})
+
+    if settings.minio_server:
+        minio_config.update({"endpoint": settings.minio_server})
+
+    if settings.s3_credentials_provider == "minio":
+        logger.info(
+            f"S3_Credentials provider - {settings.s3_credentials_provider}"
+        )
+        minio_config.update(
+            {
+                "access_key": settings.minio_root_user,
+                "secret_key": settings.minio_root_password,
+            }
+        )
+    elif settings.s3_credentials_provider == "aws_iam":
+        logger.info(
+            f"S3_Credentials provider - {settings.s3_credentials_provider}"
+        )
+        minio_config.update({"credentials": IamAwsProvider()})
+    elif settings.s3_credentials_provider == "aws_env":
+        logger.info(
+            f"S3_Credentials provider - {settings.s3_credentials_provider}"
+        )
+        minio_config.update({"credentials": EnvAWSProvider()})
+    elif settings.s3_credentials_provider == "aws_config":
+        # environmental variable AWS_PROFILE_NAME should be set
+        logger.info(
+            f"S3_Credentials provider - {settings.s3_credentials_provider}"
+        )
```

Comment:
Let's move logging to a single line (L58)

Reply:
agreed